app/services/music_import_service.rb

```diff
@@ -288,7 +288,11 @@ def create_logical_structure(output, buffered_change_set_persister)
         st << { nodes: nodes, label: date }
       end
       change_set = DynamicChangeSet.new(output)
-      change_set.logical_structure = [{ label: "By Date", nodes: structure }]
+      if change_set.logical_structure[0][:label] == ["By Date"]
```

Comment:
Can you add a comment explaining why this check is made? I'm not sure what makes this label special or what other labels there may be.

Reply:
It seems like it's just a matter of whether there's already a value?